solutions/devsprint-matheus-reis-2/DeliveryAppChallenge/Service/DeliveryApiError.swift

```diff
@@ -0,0 +1,16 @@
+//
+//  DeliveryApiError.swift
+//  DeliveryAppChallenge
+//
+//  Created by pedro tres on 17/05/22.
+//
+
+import Foundation
+
+enum DeliveryApiError: Error {
+    case invalidURL
+    case errorGeneric(description: String)
```

Comment:
Nesse caso, olhando o seu código, o erro genérico não seria tão genérico assim, pois ele ocorre durante o processo de requisição. Portanto, seria interessante mudar para `requestError`.
```suggestion
    case requestError(description: String)
```